jetty-ee10/jetty-ee10-osgi/jetty-ee10-osgi-boot/src/main/java/org/eclipse/jetty/ee10/osgi/boot/OSGiMetaInfConfiguration.java

```diff
@@ -285,28 +285,33 @@ private List<Resource> getBundleAsResource(ResourceFactory resourceFactory, Bund
         File file = BundleFileLocatorHelperFactory.getFactory().getHelper().getBundleInstallLocation(bundle);
         if (file.isDirectory())
         {
+            // Add directory bundle as a directory
+            resources.add(resourceFactory.newResource(file.toPath()));
+
             for (File f : file.listFiles())
             {
                 if (FileID.isJavaArchive(f.getName()) && f.isFile())
                 {
-                    resources.add(resourceFactory.newResource(f.toPath()));
+                    // add *.jar as jar files
+                    resources.add(resourceFactory.newJarFileResource(f.toPath().toUri()));
                 }
                 else if (f.isDirectory() && f.getName().equals("lib"))
                 {
-                    for (File f2 : file.listFiles())
+                    for (File libFile : file.listFiles())
                     {
-                        if (FileID.isJavaArchive(f2.getName()) && f2.isFile())
+                        if (FileID.isJavaArchive(libFile.getName()) && libFile.isFile())
                         {
-                            resources.add(resourceFactory.newResource(f.toPath()));
+                            // add lib/*.jar as jar files
+                            resources.add(resourceFactory.newJarFileResource(f.toPath().toUri()));
                         }
                     }
                 }
             }
-            resources.add(resourceFactory.newResource(file.toPath())); //TODO really???
```

Comment:
Yeah, I doubt we need the directory that is the bundle installation directory as a Resource.